libs/pipelines/api_pipeline.py

```diff
@@ -0,0 +1,106 @@
+from collections import namedtuple
+import logging
+
+from libs.enums import Intervention, AggregateLevel
+from libs import validate_results
+from libs import build_processed_dataset
+from libs import dataset_deployer
+from libs.us_state_abbrev import US_STATE_ABBREV
+
+from libs.datasets import results_schema as rc
+from libs.functions import generate_api as api
+
+logger = logging.getLogger(__name__)
+PROD_BUCKET = "data.covidactnow.org"
+
+APIPipelineProjectionResult = namedtuple(
+    "APIPipelineProjectionResult", ["intervention", "aggregate_level", "projection_df",]
+)
+
+APIPipelineAPIRow = namedtuple("APIPipelineAPIRow", ["key", "api"])
+
+APIPipelineResult = namedtuple("APIPipelineResult", ["api_tuples"])
+
+
+def run_projections(
+    input_file, aggregation_level, intervention: Intervention, run_validation=True
+) -> APIPipelineProjectionResult:
+    """Run the projections for the given intervention for states 
+    in order to genereate the api. 
+
+    Args:
+        input_file: Input file to load model output results from.
+        intervention: Intervention Enum to be used to genreate results
+        run_validation: If true runs validation on generated shapefiles
+            and dataframes.
+
+    Returns: APIPipelineProjectionResult objects for county data.
+    """
+
+    if aggregation_level == AggregateLevel.STATE:
+        states_key_name = f"states.{intervention.name}"
+        states_df = build_processed_dataset.get_usa_by_states_df(
+            input_file, intervention.value
+        )
+        if run_validation:
+            validate_results.validate_states_df(states_key_name, states_df)
+
+        state_results = APIPipelineProjectionResult(
+            intervention, AggregateLevel.STATE, states_df
+        )
+        return state_results
+    elif aggregation_level == AggregateLevel.COUNTY:
+        # Run County level projections
+        counties_key_name = f"counties.{intervention.name}"
+        counties_df = build_processed_dataset.get_usa_by_county_with_projection_df(
+            input_file, intervention.value
+        )
+        if run_validation:
+            validate_results.validate_counties_df(counties_key_name, counties_df)
+
+        county_results = APIPipelineProjectionResult(
+            intervention, AggregateLevel.COUNTY, counties_df
+        )
+
+        return county_results
+    else:
+        raise Exception("Non-valid aggreation level specified")
```

Comment:
I prefer `ValueError` errors here - just a bit more specific (also typo in aggregation)